### DISCUSSION TITLE
Trouble Using in Alpine Linux (Docker)

### DISCUSSION THREAD

**Discussion by jdholtz** [seleniumbase/SeleniumBase#2111] [Q&A] [answered]
Hey. I am getting the following error when attempting to run SeleniumBase in Alpine Linux through Docker: `FileNotFoundError: [Errno 2] No such file or directory: '/root/.cache/selenium/chromedriver/linux64/117.0.5938.62/chromedriver'`.

This file does exist, but when I try to execute it manually, it gives the same error (not found). This appears to be that SB is downloading the file wrong (or for the wrong architecture? - the file command says it is x86_64 though). Is Alpine supported or am I getting this error because Alpine is specifically not supported for this project? I think it would be helpful to people trying to build a minimal Docker image with SB as Alpine Linux saves a lot of space in the final image. 

Here is a minimal Dockerfile that reproduces the issue:
```Dockerfile
FROM python:3.11-alpine

WORKDIR /app

RUN apk update && apk add --no-cache chromium

RUN pip3 install --upgrade pip && pip3 install --no-cache-dir seleniumbase

ENTRYPOINT ["python", "-c", "from seleniumbase import Driver; Driver()"]
```

**Comment by mdmintz** [comment, marked as answer]
SeleniumBase doesn't touch the `.cache/` folder... That folder is specific to raw selenium driver & browser downloads. SeleniumBase may try to use existing drivers found on the PATH (which may include `.cache/`), but if something from there generates errors, then make sure you have the drivers you need already in the `seleniumbase/drivers` folder.

You can get drivers in there via:

```bash
sbase get chromedriver

sbase get uc_driver
```

As for supported architecture, UC Mode requires non-ARM drivers. ARM is fine for regular Chrome, but UC Mode will need the non-ARM version in order to properly patch the driver.

Also, if you get errors, I'll need to see a more specific stack trace in order to know the exact way an error happened. And I'll need to know the seleniumbase version used.

**Reply by jdholtz** [reply]
It actually started working once I downloaded the `chromium-chromedriver` package on SB 4.18.6. Thank you for the quick response!

**Reply by jdholtz** [reply]
As a side note, I am able to use UC with ARM (`uname -r`: `6.2.10-1-aarch64-ARCH`) if I install the above package on Alpine Linux.

**Comment by mastafadhil** [comment]
hi there @jdholtz or @mdmintz , could you help to share the working dockerfile with the steps required to "download the chromium-chromedriver package'? i recently started working on this (using alpine linux docker image) and found the same issue, previously i tried to run another project which uses ubuntu and had no trouble with it. Thank you so much in advance.

**Reply by jdholtz** [reply]
In the Docker file, you can change `RUN apk update && apk add --no-cache chromium` to `RUN apk update && apk add --no-cache chromium chromium-chromedriver` which fixed it for me. 

[Here](https://github.com/jdholtz/auto-southwest-check-in/blob/master/Dockerfile) is the Dockerfile of the project I use SeleniumBase for.
